### ISSUE TITLE
[discussion] Intermediate representation of graph in NAS

### ISSUE THREAD

**Issue by gaocegege** [kubeflow/katib#461]
Now we are using a custom IR (architecture + nn_config) to represent a candidate model, which may be insufficiently expressive, I think.

I think we should use mature IRs such as Relay IR or ONNX to represent the candidate model.

/cc @DeeperMind @andreyvelich

**Comment by gaocegege** [comment]
Ref https://github.com/dmlc/tvm/issues/1673

**Comment by andreyvelich** [comment]
@gaocegege This is make sense. What changes in API do we need to make IR like Relay IR or ONNX. And what will be the best for the NAS tasks? 
/cc @johnugeorge @richardsliu @hougangliu @YujiOshima

**Comment by gaocegege** [comment]
We need to investigate which IR works well for the case. [microsoft/nni](github.com/microsoft/nni) uses ONNX. While I think ONNX is not expressive.

@anchovYu is working on this issue.

**Comment by andreyvelich** [comment]
That is cool! Just let us know which structure you want to use and will our definition of Neural Network work for that. For RL you can find it [here](https://github.com/kubeflow/katib/blob/master/pkg/suggestion/v1alpha1/NAS_Reinforcement_Learning/README.md).

**Comment by gaocegege** [comment]
@andreyvelich Yeah, sure. Maybe we could discuss it in the katib meeting. Actually, we are not sure about the choice, too.

There are several ways to represent the search space of NAS: chain-structured (what we are using now), chain-structured with jumps and other advanced links, and cell-structured. I cannot figure a way to represent such complex space definition, even via ONNX or Relay IR or other IRs.

**Comment by andreyvelich** [comment]
Alright, let discuss at Katib meeting in this week.
/cc @DeeperMind

**Comment by gaocegege** [comment]
https://github.com/microsoft/onnxruntime/tree/master/orttraining ONNX runtime supports training ONNX. Maybe we could have a look.

/cc @andreyvelich
